cordova-lib/src/cordova/platform_metadata.js

```diff
@@ -0,0 +1,102 @@
+/**
+    Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+*/
+
+
+var path         = require('path'),
+    cordova_util = require('./util'),
+    fs           = require('fs'),
+    Q            = require('q'),
+    shelljs      = require('shelljs');
+
+
+// Retrieves the platforms and their versions from the platforms.json file
+// Returns an array of {platform: platform, version: version} ...
+// ... where version could be '3.4.0', '/path/to/platform' or 'git://...'
+function getVersions(projectRoot) {
+    var platformsDir = path.join(projectRoot, 'platforms');
+    var platformsJsonFile = path.join(platformsDir, 'platforms.json');
+
+    // If the platforms.json file doesn't exist, retrieve versions from platforms installed on the filesystem...
+    // ...Note that in this case, we won't be able to know what source(folder, git-url) the platform came from, we'll just use versions
+    return getPlatVersionsFromFile(platformsJsonFile).fail(function(){
+        return getPlatVersionsFromFileSystem(projectRoot);
+    });
+}
+
+// Returns a promise
+function getPlatVersionsFromFile(platformsJsonFile){
+    var platformData = require(platformsJsonFile);
```

Comment:
NodeJS actually doing JSON parsing when JSON file is required. See http://nodejs.org/api/modules.html#modules_file_modules and http://nodejs.org/api/modules.html#modules_all_together